### PR TITLE
Adds Covie Weapons

### PR DESCRIPTION
**Pull request by X0-11**
With fire sounds:
Needler
Plasma Pistol

Fire sounds needed:
Plasma Rifle

Plasma Pistol has a way to switch between normal and overcharge, doing increased damage.
Energy Sword has a deploy sound.

### CODE REVIEW

File: code/modules/halo/weapons/covenant/ammo.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/obj/item/projectile/covenant/plasmapistol
+	damage = 25
+	accuracy = -1
+
+/obj/item/projectile/covenant/plasmapistol/overcharge
+	damage = 75
+
+/obj/item/projectile/covenant/plasmarifle
+	damage = 30
+	accuracy = 1
```
Comment (caelaislinn) [inline code/modules/halo/weapons/covenant/ammo.dm:17]: Add placeholder sprites please

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*
+u/*
```
Comment (caelaislinn) [inline code/modules/projectiles/gun.dm]: Compile error from mis-key

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 	for(var/propname in properties)
 		var/propvalue = properties[propname]
-		
+
```
Comment (caelaislinn) [inline code/modules/projectiles/gun.dm]: Unnecessary line change

**Review by caelaislinn — CHANGES_REQUESTED** [review]

File: code/modules/halo/weapons/covenant/ammo.dm (hunk 1 above)
Comment (X0-11) [reply code/modules/halo/weapons/covenant/ammo.dm:17]: Will do. I'll do it once home.

**Comment by X0-11** [comment]
Sprites added

File: code/modules/halo/weapons/covenant/ammo.dm [hunk 4]
```diff
@@ -0,0 +1,44 @@
+ // need icons for all projectiles and magazines
+/obj/item/projectile/covenant
+	name = "Plasma Bolt"
+	desc = "A searing hot bolt of plasma."
```
Comment (caelaislinn) [inline code/modules/halo/weapons/covenant/ammo.dm:4]: I dislike leaving "ghost" objects like this in SS13. C++ calls this an abstract virtual class but in Byond we don't have any protections against other people misusing it plus it clutters up the path tree. 

What I do it instead is choose a base object (usually the simplest and most commonly reused subtype of the object) then descend children off it. This is just a coding convention though so it won't block merging

**Review by caelaislinn — APPROVED** [review]

**Pull request merged**